### PR TITLE
[Refactor] Use new API Command: UpdateBeneficialOwnersForBankAccount

### PR DESCRIPTION
**Pull request by NikkiWines**
### Details
Associated web PR: https://github.com/Expensify/Web-Expensify/pull/34853

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/226855

### Tests
- [x] Verify that no errors appear in the JS console

#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps

1. In NewExpensify log into an account with a workspace / Create an account and then create a workspace
2. Navigate to Settings > Workspace > Connect Bank Account 
3. Follow the steps from [this StackOverflow post](https://stackoverflow.com/c/expensify/questions/342) to add a **verifying** bank account
4. Confirm that after submitting the beneficial owners step that you are routed to the `Validate Bank Account - Let's finish in chat` page

Screen Shot 2022-10-17 at 4 23 46 PM

#### Web
**Pending:**

https://user-images.githubusercontent.com/16747822/193106725-c631d34d-29bb-4d4d-92b8-0ee5c6364488.mov

**Verifying:**

https://user-images.githubusercontent.com/16747822/193106767-70fbfc2a-2119-4958-acb0-5bb8de3794cc.mov

**Open**

https://user-images.githubusercontent.com/16747822/194065307-61e58fc9-e8cb-4466-a19e-600e18d63204.mov

#### Desktop

https://user-images.githubusercontent.com/16747822/194864495-9da38ca7-4cb6-4782-ba33-09975bbe4a8d.mov

#### mWeb 

https://user-images.githubusercontent.com/16747822/194864521-08f302bf-d0dc-4cb1-a0ac-34984d536749.mp4

#### iOS

https://user-images.githubusercontent.com/16747822/194895039-681fbe75-4600-4a77-9772-b78868c1726c.mp4

#### Android

[android-pending.webm](https://user-images.githubusercontent.com/16747822/194895068-c8a1cc2c-4d90-40c0-be7b-ab75f1005546.webm)

### CODE REVIEW

**Comment by NikkiWines** [comment]
Oops, sorry @sketchydroide - please hold off reviewing for now, this isn't quite ready

**Comment by nkuoch** [comment]
https://github.com/Expensify/App/pull/11162 has been merged on main - please merge main on your branch

**Review by MariaHCD — COMMENTED** [review]
LGTM! We'll just need the test steps and QA steps added for this one

**Review by MariaHCD — DISMISSED** [review]

**Review by sketchydroide — DISMISSED** [review]
LGTM

**Comment by sketchydroide** [comment]
Do we need to add C+?

**Comment by NikkiWines** [comment]
@sketchydroide I don't think it has to be a C+ to do the PR reviewer checklist, any reviewer should be able to check things so that the test passes?

**Comment by JmillsExpensify** [comment]
We've been talking about this topic in this #whatsnext [post](https://expensify.slack.com/archives/CC7NECV4L/p1665064570315429). As a quick summary, this is what's being proposed as the new standard:

> - PRs that change code that runs in any of our platforms must have screenshots/videos (proof of testing) for all platforms (same as today for contributors, but a new standard for us)
> - Both checklists (author & reviewer) must be filled out on all PRs (same as today for contributors, but a new standard for us)
> - C+ is assigned to all PRs (same as today for contributors, but a new standard for us)

**Comment by NikkiWines** [comment]
Added videos of testing on all platforms and assigned a C+

File: src/pages/ReimbursementAccount/reimbursementAccountPropTypes.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ export default PropTypes.shape({
     error: PropTypes.string,
 
     /** Which field needs attention? */
-    errorFields: PropTypes.objectOf(PropTypes.bool),
+    errorFields: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.bool, PropTypes.array])),
```
Comment (NikkiWines) [inline src/pages/ReimbursementAccount/reimbursementAccountPropTypes.js:31]: Confirmed that error types show up correctly for both types: 

**Boolean**
Screen Shot 2022-10-10 at 17 00 31


**Array**

Screen Shot 2022-10-10 at 16 43 32

**Comment by NikkiWines** [comment]
Updated once more

**Comment by JmillsExpensify** [comment]
@NikkiWines ah sorry for the confusion! I don't think you can assign C+ this way in the PR directly. In order to keep this PR moving, it's probably best if someone in this PR take on the testing steps that remain, what do you think? 

Then for clarity, it looks like you only provided a screenshot of a single mobile platform. Doesn't the PR template specifically prompt for two mobile web platforms (chrome on Android and safari on iOS)?

**Comment by NikkiWines** [comment]
> @NikkiWines ah sorry for the confusion! I don't think you can assign C+ this way in the PR directly. In order to keep this PR moving, it's probably best if someone in this PR take on the testing steps that remain, what do you think?

Oh, sorry - I assumed the `contributor-plus` assignment would assign a C+ but looks like that's not the case 😅. I think it's fine if one of the 3 current reviewers takes on the testing steps. 

> Then for clarity, it looks like you only provided a screenshot of a single mobile platform. Doesn't the PR template specifically prompt for two mobile web platforms (chrome on Android and safari on iOS)?

Oh, yep - good catch. Let me add mWeb for Android now.

**Comment by Santhosh-Sellavel** [comment]
@NikkiWines 
Not sure I believe the process is to create a issue [example](https://github.com/Expensify/App/issues/11067) and add [ContributorPlusReview](https://github.com/Expensify/App/labels/ContributorPlusReview) label.

**Comment by NikkiWines** [comment]
The flow works fine for mWeb Android - Chrome but some of the UI elements in views not impacted by this PR (Company and Requestor steps) aren't displaying correctly. The `zipCode` and `first name` fields successfully receive input but don't display it correctly. 

[mWeb - chrome.webm](https://user-images.githubusercontent.com/16747822/194916572-07b4b5cb-906d-4cf3-aa8c-59989d85ab33.webm)

@sketchydroide @MariaHCD could one of you confirm if this is also the case for your sims (or on an actual android device)? 

Also would one of you take on the reviewer testing steps, please?

**Review by MariaHCD — APPROVED** [review]

**Comment by MariaHCD** [comment]
- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Comment by MariaHCD** [comment]
Going to test this on all platforms (hopefully, I don't run into any issues with the simulators 😅)

**Comment by MariaHCD** [comment]
1. Testing on the iOS simulator

@NikkiWines Have you tested checking the boxes for `I own more than 25%..` and/or `Somebody else owns more than 25%...`? I checked those boxes for an `OPEN` bank account and landed on an empty ValidationStep page:

https://user-images.githubusercontent.com/12268372/195094239-2512cef9-3c48-4f22-ac32-33d0ecc41498.mov

I also tried the flow without checking those two boxes and still ended up on an empty ValidationStep page 😕 

I also get this error pop up during the setup process (but I don't think it's related to this PR) 🤔 
 
Simulator Screen Shot - iPhone 14 - 2022-10-11 at 16 40 15

Simulator Screen Shot - iPhone 14 - 2022-10-11 at 16 51 34

**Comment by MariaHCD** [comment]
2. Testing on MacOS / Chrome

Hmm, getting an error after selecting a Plaid bank account. Also doesn't seem related to this PR 🤔 

Screen Shot 2022-10-11 at 5 44 52 PM

**Comment by NikkiWines** [comment]
> @NikkiWines Have you tested checking the boxes for I own more than 25%.. and/or Somebody else owns more than 25%...? I checked those boxes for an OPEN bank account and landed on an empty ValidationStep page:

Also able to reproduce this on Web - digging into it now. 

> I also tried the flow without checking those two boxes and still ended up on an empty ValidationStep page 😕

Weird, I'm getting this now too but it was definitely working before 🤔 

> Hmm, getting an error after selecting a Plaid bank account. Also doesn't seem related to this PR 🤔

I'm also getting these now - maybe some new code merged to main?

**Comment by NikkiWines** [comment]
For the blank validationStep flows it looks like using the `Wells Fargo` creds and following the [SO post](https://stackoverflow.com/c/expensify/questions/342/525#525) is returning the bank account in state `SETUP`? No idea what would've caused that regression but will continue digging.

**Comment by NikkiWines** [comment]
Seemed to be working briefly after pulling main on auth and rebuilding but now i'm getting the blank `Test Transactions` view once more. Will continue digging tomorrow

**Review by MariaHCD — COMMENTED** [review]
@NikkiWines Thanks for investigating that! It's probably due to this fix for VerifyIdentityForBankAccount: https://github.com/Expensify/Web-Expensify/pull/35108

We started sending `useOnfido` as true for every step in the flow [here](https://github.com/Expensify/Web-Expensify/blob/684154d3c2d1b96a54da1ab4388b42861653ed20/lib/BankAccountAPI.php#L1123-L1132). And we actually don't need `requestorIdentityKBA` anymore. It used to be a fallback to Onfido but we removed this fallback for NewDot (more details [here](https://docs.google.com/document/d/1PgmgfTWBVIl3tGDui8HgSNezgCrSyuS_We-msd7PJag/edit#bookmark=id.l2oluo2v4j4n))

So it sounds like we just need to understand when Onfido can be skipped and when it cannot. 

cc: @nkuoch

**Comment by JmillsExpensify** [comment]
Question question: with 34853 merged and now on production can we remove the [HOLD] on this PR?

**Comment by MariaHCD** [comment]
We still need to work through a small issue with the flow: https://github.com/Expensify/App/pull/10994#issuecomment-1275943147

**Comment by MariaHCD** [comment]
Looks like we get the error `(BankAccount.cpp:2385) isSetupComplete [socket298] [info] Setup is not complete because externalAPIResponses['requestorIdentityKBA'] is missing` because useOnfido is false [here](https://github.com/Expensify/Auth/blob/bd28798f72440d41d863ac4f29b16a1b72b373a4/auth/lib/BankAccount.cpp#L2367) (the bank account's additionalData does not have the `onfidoData`)

I'm not 100% sure of why/when we skip Onfido during the VBBA flow, I can dig into it more tomorrow

**Review by sketchydroide — APPROVED** [review]

**Comment by MariaHCD** [comment]
Made sure to pull the latest changes from Auth and the tests work better today 🎊 

 1. iOS / native 
 
https://user-images.githubusercontent.com/12268372/195596649-5c62ced0-5c7d-437f-8954-e8c0bb2e3208.mov
 
 2. Android / native
 
https://user-images.githubusercontent.com/12268372/195625458-24987c0a-5e53-4950-9353-cb233a220339.mov
 
 3. iOS / Safari

https://user-images.githubusercontent.com/12268372/195599091-8b6f290a-1411-4e41-b453-631adc0c9e47.mov
 
 4. Android / Chrome
 
https://user-images.githubusercontent.com/12268372/195628895-0226c443-b4de-4c80-af7b-269b5df3249c.mp4


 5. MacOS / Chrome
 
[(2) New Expensify.webm](https://user-images.githubusercontent.com/12268372/195592843-cae7e7ca-83a9-4671-a177-7f3388bf721a.webm)

 6. MacOS / Desktop

https://user-images.githubusercontent.com/12268372/195605198-32a6866e-be3e-4409-990b-800176aafe25.mov

https://user-images.githubusercontent.com/12268372/195605172-0d3b69f8-4778-4f81-bf78-c6fbd5017328.mov

**Comment by NikkiWines** [comment]
Thanks for all the testing @MariaHCD!! Went through it again myself and got the same results as you. 

It looks like things may have just changed with https://github.com/Expensify/Auth/pull/7091 just getting merged but to recap this is why the `OPEN` VBBA flow is routing to the `SETUP` state instead of `OPEN`. 


**TLDR;**  we need to make sure `useOnfido` is defined by the same check in `getFilteredAdditionalDataJSON()` [here](https://github.com/Expensify/Auth/blob/1630ba48e903dd9c232365ad895876c84b159409/auth/lib/BankAccount.cpp#L1660) and in `isSetupComplete()` [here](https://github.com/Expensify/Auth/blob/1630ba48e903dd9c232365ad895876c84b159409/auth/lib/BankAccount.cpp#L2152).  Right now that bool `true` in `getFilteredAdditionalDataJSON()` but `false` in `isSetupComplete()` - which results in our error and bad setup state. 

<details closed>
<summary><b>Long explanation</b></summary>
<br>

We pass along both `useOnfido: '1' ` and `requestorIdentityKBA: '[status: 'pass' apiResult: '[reason: 'Tested via Onfido instead']'` in the additional data when calling the `SetupWithdrawalAccount` command (when setting up with the Wells Fargo Plaid creds)

In `getFilteredAdditionalDataJSON()` we [define](https://github.com/Expensify/Auth/blob/1630ba48e903dd9c232365ad895876c84b159409/auth/lib/BankAccount.cpp#L1660) `const bool useOnfido` based only on if `useOnfido` is true or not when it's passed to the command. In that function we call `getRequiredVerificationSetupAttributes()` which takes in `useOnfido`. Because `useOnfido` is true at this point, our step name to use is `ACHContractStepWithOnfido` which returns [nothing](https://github.com/Expensify/Auth/blob/dcd6310fa983ee8128de735348b3f7b8f561e8a2/auth/lib/BankAccount.cpp#L117) for our required setup attributes. 

This results in us leaving out `requestorIdentityKBA` from the `externalApiResponses` that we add to the `additionalDataJSON`. 

Then, later in `isSetupComplete()`, `const bool useOnfido` is based off of both `useOnfido` being true, _and_ `onfidoData` being passed (as @MariaHCD mentioned [here](https://github.com/Expensify/App/pull/10994#issuecomment-1276336787). Because we're not passing `onfidoData`, `useOnfido` is now false and our step name to use is `ACHContractStep` returns an [`requestorIdentityKBA`](https://github.com/Expensify/Auth/blob/dcd6310fa983ee8128de735348b3f7b8f561e8a2/auth/lib/BankAccount.cpp#L116) as a key for our required setup attributes. Since that's now required - we end up failing and returning [here](https://github.com/Expensify/Auth/blob/c507e358f31debe497dc1abb512f4bc8a6af4507/auth/lib/BankAccount.cpp#L1083) leaving our bank account in state `setup`. 

</details>

**Comment by JmillsExpensify** [comment]
Quick question: Are we holding on this PR until the Auth PR for consistently handling Onfido is merged?

**Comment by MariaHCD** [comment]
Thinking about @JmillsExpensify's [comment](https://github.com/Expensify/App/pull/10994#issuecomment-1280679532), I think we can merge this PR. The bug with the flow is on the backend. I've created a separate issue to investigate and fix that: https://github.com/Expensify/Expensify/issues/234337

**Comment by JmillsExpensify** [comment]
Nice, agree with that. Thanks all!

**Comment by roryabraham** [comment]
Hi everyone 👋🏼 

It seems like this PR is the most likely culprit for [this deploy blocker](https://github.com/Expensify/App/issues/11929). Would be great if anyone with context on the changes in this PR could look into that 🙇🏼

**Comment by MariaHCD** [comment]
> It seems like this PR is the most likely culprit for [this deploy blocker](https://github.com/Expensify/App/issues/11929). Would be great if anyone with context on the changes in this PR could look into that 🙇🏼

This is likely a backend error and not a frontend error. We're investigating here: https://github.com/Expensify/Expensify/issues/234337
